From 2335b4060b046edc23609a4cac6e4c8cc4641687 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 30 Jun 2022 14:16:32 +0300
Subject: [PATCH] [SW-93513] gaudi2: add security configuration in erom recovery

This patch adds security configurations which needs to be enabled
before bringing up pcie link in erom_pcie and in erom_recovery in
a secured system.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I707efb123eff11be2f2f98c2afa1e9487fb79fe0
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 7c591f5..1145dfc 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -56,6 +56,7 @@
 	$(PREBOOT_ROOT_DIR)/lib/common/spi.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/spi_flash.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/lib_common.o \
+	$(HL_DEV_ARCH)/../common/security.o \
 	$(HL_DEV_ARCH)/$(HL_TARGET).o \
 	$(HL_DEV_ARCH)/../common/$(HL_TARGET).o \
 	$(HL_DEV_ARCH)/../common/common.o \
diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index 3008bf7..905db81 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -238,12 +238,19 @@
 	return 0;
 }
 
+__weak void ppboot_security_init(void) {}
+__weak void ppboot_secure_arc(void) {}
+
 void erom_main_function(void)
 {
 	void (*ppboot)();
 	int ret;
 
 	device_init();
+
+	/* Set ARCs to privileged mode */
+	ppboot_secure_arc();
+
 	/*
 	 * Note: we assume CPU frequency is not changed while CPU is running,
 	 * so the udelay init is done once before we start.
@@ -268,5 +275,6 @@
 
 	cpu_boot_if_init();
 	hl_plat_early_init();
+	ppboot_security_init();
 	erom_pcie_init();
 }
diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index ea25bcc..5d1f4e4 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -119,6 +119,10 @@
 
 	preboot_device_id_init();
 
+	preboot_security_init();
+	preboot_security_dynamic_init();
+	cpu_boot_sts_set(CPU_BOOT_STATUS_SECURITY_READY);
+
 	hl_ts_init();
 
 	/* Prepare boot FW versions and eFuse data for host */
